compiler/luci/pass/src/FuseMulToFullyConnectedWeightsPass.test.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseMulToFullyConnectedWeightsPass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+#include <luci/test/TestIOGraph.h>
+
+#include <gtest/gtest.h>
+
+namespace
+{
+
+using namespace luci::test;
+
+template <loco::DataType DT> class FuseMulToFullyConnectedWeightsPassTestGraph : public TestIOGraph
+{
+public:
+  FuseMulToFullyConnectedWeightsPassTestGraph() = default;
+
+  void init(void)
+  {
+    TestIOGraph::init({3, 4}, {3, 6});
+
+    _mul = g()->nodes()->create<luci::CircleMul>();
+    _mul_s = g()->nodes()->create<luci::CircleConst>();
+    _fc = g()->nodes()->create<luci::CircleFullyConnected>();
+    _fc_w = g()->nodes()->create<luci::CircleConst>();
+    _fc_b = g()->nodes()->create<luci::CircleConst>();
+
+    _mul->name("mul");
+    _mul_s->name("mul_s");
+    _fc->name("fc");
+    _fc_w->name("fc_w");
+    _fc_b->name("fc_b");
+
+    _mul->dtype(DT);
+    _fc->dtype(DT);
+    _mul->fusedActivationFunction(luci::FusedActFunc::NONE);
+    _fc->fusedActivationFunction(luci::FusedActFunc::NONE);
+
+    _mul_s->rank(1);
+    _mul_s->dim(0) = 3;
+    _mul_s->dtype(DT);
+    _mul_s->size<DT>(3);
+    for (uint32_t i = 0; i < 3; ++i)
+    {
+      _mul_s->at<DT>(0) = 1.0f;
+    }
+
+    _fc_w->rank(2);
+    _fc_w->dim(0) = 3;
+    _fc_w->dim(1) = 4;
+    _fc_w->dtype(DT);
+    _fc_w->size<DT>(4 * 6);
+    for (uint32_t i = 0; i < 4 * 6; ++i)
```

Comment:
The shape of `_fc_w` is <3x4>
Does this size have to be the same to the shape..?

```suggestion
    _fc_w->dim(0) = 3;
    _fc_w->dim(1) = 4;
    _fc_w->dtype(DT);
    _fc_w->size<DT>(3 * 4);
    for (uint32_t i = 0; i < 3 * 4; ++i)
```

Reply:
or

```suggestion
    _fc_w->dim(0) = 4;
    _fc_w->dim(1) = 6;
    _fc_w->dtype(DT);
    _fc_w->size<DT>(4 * 6);
    for (uint32_t i = 0; i < 4 * 6; ++i)
```

Reply:
please check about FullyConnected Op.

with torch,
```py
import torch

tensor1 = torch.randn(3, 4)
FC = torch.nn.Linear(4, 6)
output = FC(tensor1)
print(output)
print(output.shape)
```
gives something like
```
tensor([[-0.8929,  0.2467, -0.1407, -0.9917, -0.3787,  0.6504],
        [-0.2442, -0.0493, -0.1175, -1.0433,  0.0100, -0.3347],
        [ 0.7240,  0.0195, -0.3735, -1.5789, -0.0348, -0.9252]],
       grad_fn=<AddmmBackward0>)
torch.Size([3, 6])
```

Reply:
more simple case is with vector,
```py
tensor1 = torch.randn(4)
FC = torch.nn.Linear(4, 6)
output = FC(tensor1)
print(output)
print(output.shape)
```
gives
```
tensor([ 0.7355, -0.1311,  0.4749, -0.1821, -0.1245, -1.0594],
       grad_fn=<ViewBackward0>)
torch.Size([6])
```

Reply:
ah, should be
```
    _fc_w->dim(0) = 6;
    _fc_w->dim(1) = 4;
```